src/Lean/Data/Lsp/Communication.lean

```diff
@@ -117,6 +117,24 @@ section
 
   def writeLspResponseErrorWithData (h : FS.Stream) (e : ResponseError α) : IO Unit :=
     h.writeLspMessage e
+
+  partial def lspShutdown (hOut : FS.Stream) (hIn : FS.Stream) (requestNo : Nat) : IO Unit := do
```

Comment:
Let's put this in `Ipc.lean` as it is only used for testing the server, and let's put a docstring explaining that it performs the LSP shutdown sequence.

Reply:
done.